src/components/SlowPageAlert/index.tsx

```diff
@@ -23,24 +21,29 @@ const SlowPageAlertBase = ({
   allowSlowPagesByDefault,
   getLinkText,
   getMessage,
+  history,
   location,
 }: PublicProps) => {
   const allowSlowPages = _shouldAllowSlowPages({
     allowByDefault: allowSlowPagesByDefault,
     location,
   });
 
-  const newLocation = `${location.pathname}${createAdjustedQueryString(
-    location,
-    {
-      [allowSlowPagesParam]: !allowSlowPages,
-    },
-  )}`;
+  const onClick = () => {
+    const newLocation = {
+      ...location,
+      search: queryString.stringify({
```

Comment:
I forgot how it works but can't we use `location.query` here?

Reply:
There is no such thing as `location.query`, just `location.search`. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/history/index.d.ts#L29-L35

Reply:
oh right, in addons-frontend we patch the `location`: https://github.com/mozilla/addons-frontend/blob/43c481afa8e0ab37494f250639efc2511de93fc2/src/core/server/base.js#L56